instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java

```diff
@@ -35,101 +42,121 @@
+    result.put(RequestType.DynamoDB, new DbRequestDecorator());
+    return result;
+  }
+
+  private Map<String, String> mapFieldToAttribute() {
+    Map<String, String> result = new HashMap<>();
+    result.put("QueueUrl", "aws.queue.url");
+    result.put("Bucket", "aws.bucket.name");
+    result.put("QueueName", "aws.queue.name");
+    result.put("StreamName", "aws.stream.name");
+    result.put("TableName", "aws.table.name");
+    return result;
+  }
 
   TracingExecutionInterceptor(Kind kind) {
-    this.kind = kind;
+    this.spanFactory = new SpanFactory(kind);
+  }
+
+  private Optional<SdkRequestDecorator> decorator(ExecutionAttributes executionAttributes) {
+    RequestType type = getTypeFromAttributes(executionAttributes);
+    return Optional.ofNullable(decoratorsMap.get(type));
+  }
+
+  private RequestType getTypeFromAttributes(ExecutionAttributes executionAttributes) {
+    return executionAttributes.getAttribute(REQUEST_TYPE_ATTRIBUTE);
   }
 
   @Override
   public void beforeExecution(
       Context.BeforeExecution context, ExecutionAttributes executionAttributes) {
-    Span span = TRACER.getOrCreateSpan(spanName(executionAttributes), AwsSdk.tracer());
+    Span span = spanFactory.forExecutionAttributes(executionAttributes);
     executionAttributes.putAttribute(SPAN_ATTRIBUTE, span);
+    ofSdkRequest(context.request())
+        .ifPresent(rt -> executionAttributes.putAttribute(REQUEST_TYPE_ATTRIBUTE, rt));
   }
 
   @Override
   public void afterMarshalling(
       Context.AfterMarshalling context, ExecutionAttributes executionAttributes) {
-    Span span = executionAttributes.getAttribute(SPAN_ATTRIBUTE);
+    Span span = getSpanFromAttributes(executionAttributes);
     if (span != null) {
       TRACER.onRequest(span, context.httpRequest());
-      onSdkRequest(span, context.request());
-      onAttributes(span, executionAttributes);
+      decorator(executionAttributes)
+          .ifPresent(dec -> dec.decorate(span, context.request(), executionAttributes));
+      decorateWithGenericRequestData(span, context.request());
+      decorateWithExAttributesData(span, executionAttributes);
     }
   }
 
+  private void decorateWithGenericRequestData(Span span, SdkRequest request) {
+
+    ofSdkRequest(request)
+        .ifPresent(
+            type -> {
+              for (String field : type.getFields()) {
+                request
+                    .getValueForField(field, String.class)
+                    .ifPresent(val -> span.setAttribute(fieldToAttribute.get(field), val));
+              }
+            });
+  }
+
+  private void decorateWithExAttributesData(Span span, ExecutionAttributes attributes) {
+
+    String awsServiceName = attributes.getAttribute(SdkExecutionAttribute.SERVICE_NAME);
+    String awsOperation = attributes.getAttribute(SdkExecutionAttribute.OPERATION_NAME);
+
+    span.setAttribute("aws.agent", COMPONENT_NAME);
```

Comment:
@anuraaga do you (at AWS I mean) have plans for formalising AWS SDK semantic convention in Otel spec?

Reply:
Yeah I've started writing a markdown doc for it but need to finish it 😅 This one in particular should probably be deleted since it overlaps with the `InstrumentationLibraryInfo` but that's for a separate PR

Reply:
It has already been here (not my change) so I'll resolve the conversation until we have a semantic convention in place :)